resources/idlharness-shadowrealm.js

```diff
@@ -0,0 +1,78 @@
+// TODO: it would be nice to support `idl_array.add_objects`
+/**
+ * idl_test_shadowrealm is a promise_test wrapper that handles the fetching of the IDL, and
+ * running the code in a `ShadowRealm`, avoiding repetitive boilerplate.
+ *
+ * @see https://github.com/tc39/proposal-shadowrealm
+ * @param {String[]} srcs Spec name(s) for source idl files (fetched from
+ *      /interfaces/{name}.idl).
+ * @param {String[]} deps Spec name(s) for dependency idl files (fetched
+ *      from /interfaces/{name}.idl). Order is important - dependencies from
+ *      each source will only be included if they're already know to be a
+ *      dependency (i.e. have already been seen).
+ */
+function idl_test_shadowrealm(srcs, deps) {
+    const script_urls = [
+        "/resources/testharness.js",
+        "/resources/WebIDLParser.js",
+        "/resources/idlharness.js",
+    ];
+    promise_test(async t => {
+        const realm = new ShadowRealm();
+        realm.evaluate("globalThis.self = globalThis; undefined");
```

Comment:
Can you add a TODO here about https://github.com/web-platform-tests/wpt/pull/31851/files#r761766428?